### PR TITLE
Refresh upstream

### PR DESCRIPTION
**Pull request by kwmonroe**
Sync/bump our default versions with latest upstream manifests.  Couple mypy fixes and some urllib love since it hates downloading `Path()` real bad.

Depends on https://github.com/charmed-kubernetes/bundle/pull/875

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,8 +41,8 @@ kubectl describe nodes |egrep "Taints:|Name:|Provider"
 ### Details
 
 * Requires a `charmed-kubernetes` deployment on a vsphere cloud launched by juju with the `allow-privileged` flag enabled.
-* Deploy the `vsphere-integrator` charm into the model using `--trust` so juju provided vsphere credentials
-* Deploy the `vsphere-cloud-provider` charm in the model relating to the integrator and to charmed-kubernetes components
+* Deploy the `vsphere-integrator` charm into the model with `--trust` to use juju provided vsphere credentials.
+* Deploy the `vsphere-cloud-provider` charm and relate to the integrator and charmed-kubernetes components.
```
Comment (addyess) [inline README.md:45]: no trouble here

Comment (addyess) [inline README.md:45]: no trouble here

File: upstream/cloud_provider/manifests/v1.24/vsphere-cloud-controller-manager.yaml [hunk 2]
```diff
@@ -236,7 +236,7 @@ spec:
       priorityClassName: system-node-critical
       containers:
         - name: vsphere-cloud-controller-manager
-          image: gcr.io/cloud-provider-vsphere/cpi/release/manager:v1.24.0
+          image: gcr.io/cloud-provider-vsphere/cpi/release/manager:v1.24.2
```
Comment (addyess) [inline upstream/cloud_provider/manifests/v1.24/vsphere-cloud-controller-manager.yaml:239]: nice!

File: upstream/update.py [hunk 3]
```diff
@@ -197,8 +199,8 @@ def download(source: str, release: Release) -> Release:
     manifest = SOURCES[source]["manifest"]
     dest = FILEDIR / source / "manifests" / release.name / manifest
     dest.parent.mkdir(exist_ok=True)
-    urllib.request.urlretrieve(release.path, dest)
-    return Release(release.name, str(dest), release.size)
+    urllib.request.urlretrieve(release.upstream, dest)
+    return Release(release.name, dest, release.size)
```
Comment (addyess) [inline upstream/update.py:203]: it looks like something broke down and needed repair in this script?

**Review by addyess — APPROVED** [review]
LGTM -- i'm curious what fell apart in the update job?

File: upstream/update.py (hunk 3 above)
Comment (kwmonroe) [reply upstream/update.py:203]:
```python
>>> thisisfine = Path('https://ohno.com/bazinga')
>>> try:
...     urllib.request.urlretrieve(thisisfine)
... except Exception as e:
...     print('OH NO! %s' % e)
...     print('you went for {}'.format(thisisfine))
...     print(f'but see? Path ate your slashy bit!')
... 
OH NO! expected string or bytes-like object
you went for https:/ohno.com/bazinga
but see? Path ate your slashy bit!
>>> 
```

**Pull request merged**